cmd/snap-confine/mount-support.c

```diff
@@ -743,3 +743,45 @@ void sc_setup_user_mounts(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 	sc_do_mount("none", "/", NULL, MS_REC | MS_SLAVE, NULL);
 	sc_call_snap_update_ns_as_user(snap_update_ns_fd, snap_name, apparmor);
 }
+
+void sc_ensure_snap_dir_shared_mounts(void)
+{
+	const char *dirs[] = { SNAP_MOUNT_DIR, "/var/snap", NULL };
+	for (int i = 0; dirs[i] != NULL; i++) {
+		const char *dir = dirs[i];
+		if (!is_mounted_with_shared_option(dir)) {
```

Comment:
Separately from this discussion it would be good to split the condition into a pair:

- is this a mount point
- what kind of sharing does it have

So that we can only apply one of the two operations if necessary. We got lucky this time but having this code more correct for later would be good.